java-squid/src/main/java/org/sonar/java/se/ProgramState.java

```diff
@@ -253,6 +265,22 @@ public void accept(SymbolicValue value, Object valueConstraint) {
     return result;
   }
 
+  public List<ConstrainedValue> getValuesWithConstraints(final Object state) {
```

Comment:
Why not simply returning a map  here ? value->constraint ? this would avoid to create a new object (thus a new concept).